refactor: Remove the beta endpoint for Copilot usage

henriklundstrom opened the pull request:
Removes support for the beta endpoint for Copilot usage metrics, which is scheduled for removal by GitHub by the end of the year, see https://github.blog/changelog/2024-10-30-github-copilot-metrics-api-ga-release-now-available/. Support for the GA endpoint was implemented by https://github.com/google/go-github/commit/717e93fa2808da46013f98feb95275e3f6bedf2b.

gmlewis commented:
@henriklundstrom - I'm confused - wasn't GitHub's OpenAPI spec updated to include your new endpoints?
Should I be updating the OpenAPI spec again?  (I think I updated it two days ago maybe in #3352.)

henriklundstrom commented:
> @henriklundstrom - I'm confused - wasn't GitHub's OpenAPI spec updated to include your new endpoints? Should I be updating the OpenAPI spec again? (I think I updated it two days ago maybe in #3352.)

@gmlewis There are two sets of endpoints for Copilot metrics. They do largely the same things and both of them were launched very recently. First there was a beta launch, and then shortly thereafter a GA version to replace it.

Beta:
The **_usage_** endpoints, see https://docs.github.com/en/rest/copilot/copilot-usage?apiVersion=2022-11-28. They are in public preview, and as far as I understand it from this announcement https://github.blog/changelog/2024-10-30-github-copilot-metrics-api-ga-release-now-available/, will be discontinued by the end of the year. This PR removes support for those endpoints, which was originally added about a month ago with https://github.com/google/go-github/commit/5e806b6a7e3f31f93d5e7b6225cd60f1d71d0c5a.

GA:
The **_metrics_** endpoints, see https://docs.github.com/en/rest/copilot/copilot-metrics?apiVersion=2022-11-28. They are in GA, were introduced very recently, after the **_usage_** endpoints, and as far as I understand it, are intended to replace them. These are the endpoints I implemented support for a few days ago with https://github.com/google/go-github/commit/717e93fa2808da46013f98feb95275e3f6bedf2b, and they will remain after this PR.

henriklundstrom commented:
@gmlewis I should clarify as well that yes this is a breaking change in the sense that it removes the endpoints added by https://github.com/google/go-github/commit/5e806b6a7e3f31f93d5e7b6225cd60f1d71d0c5a, but that commit has not yet made it to any release, so in that sense it is **not** breaking with any previous **release**.

gmlewis commented:
@henriklundstrom - thank you for the excellent explanation. That makes perfect sense.

gmlewis reviewed (approved):
Thank you, @henriklundstrom !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

gmlewis commented on github/copilot.go:
```diff
@@ -498,110 +465,6 @@ func (s *CopilotService) GetSeatDetails(ctx context.Context, org, user string) (
 	return seatDetails, resp, nil
 }
 
-// GetOrganizationUsage gets daily breakdown of aggregated usage metrics for Copilot completions and Copilot Chat in the IDE across an organization.
-//
-// GitHub API docs: https://docs.github.com/rest/copilot/copilot-usage#get-a-summary-of-copilot-usage-for-organization-members
-//
-//meta:operation GET /orgs/{org}/copilot/usage
```
I'm fine with deleting since the removed code has not been officially released in this repo with any tagged release.

That's why I removed the Breaking Change label.

alexandear replied:
Maybe it's better to remove "BREAKING CHANGE" from the PR's description. This will avoid any confusion in for the future PR's readers.

<img width="941" alt="image" src="https://github.com/user-attachments/assets/bdef6261-b5cc-4579-b681-c74772add385">

henriklundstrom replied:
I updated the description, hopefully it's more clear now.

gmlewis commented:
Thank you, @alexandear !
Merging.
